core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -745,7 +746,7 @@ boolean isResponseTimedOut(TaskDef taskDefinition, TaskModel task) {
             return false;
         }
 
-        if (task.getStatus().isTerminal() || isAyncCompleteSystemTask(task)) {
+        if (task.getStatus().isTerminal() || task.getTaskType().equals(SUB_WORKFLOW.name())) {
```

Comment:
@manan164 The contract for the async complete system tasks is that the response for this task will be sent asynchronously (either via API or events). For such tasks, response time out will not apply. 

However, the task timeout will still be applied and is enforced as per the `checkTaskTimeout` method. 

Could you please provide more info about this change?